WeightProducer/python/MCSampleValues.py

```diff
@@ -63,493 +63,1039 @@ class MCSampleValuesHelper():
         },
         "QCD_Pt-120to170_MuEnrichedPt5" : {
-            "CrossSection" : XSValues(XS_13TeV=21230.0),
+            "CrossSection" : XSValues(
+                XS_2016=25470, XSSource_2016="GenXSecAnalyzer",
+                XS_2017=21230, XSSource_2017="XSDB",
+                XS_2018=21190, XSSource_2018="GenXSecAnalyzer",
+            ),
         },
         "QCD_Pt-170to300_MuEnrichedPt5" : {
-            "CrossSection" : XSValues(XS_13TeV=7055.0),
+            "CrossSection" : XSValues(
+                XS_2016=8635, XSSource_2016="GenXSecAnalyzer",
+                XS_2017=7055, XSSource_2017="XSDB",
+                XS_2018=7025, XSSource_2018="GenXSecAnalyzer",
+            ),
         },
         "QCD_Pt-300to470_MuEnrichedPt5" : {
-            "CrossSection" : XSValues(XS_13TeV=797.35),
+            "CrossSection" : XSValues(
+                XS_2016=797.3, XSSource_2016="GenXSecAnalyzer",
+                XS_2017=619.8, XSSource_2017="GenXSecAnalyzer",
+                XS_2018=620.6, XSSource_2018="GenXSecAnalyzer",
+            ),
         },
         "QCD_Pt-470to600_MuEnrichedPt5" : {
-            "CrossSection" : XSValues(XS_13TeV=59.24),
+            "CrossSection" : XSValues(
+                XS_2016=79.25, XSSource_2016="GenXSecAnalyzer",
+                XS_2017=59.24, XSSource_2017="XSDB",
+                XS_2018=59.06, XSSource_2018="GenXSecAnalyzer",
+            ),
         },
         "QCD_Pt-600to800_MuEnrichedPt5" : {
-            "CrossSection" : XSValues(XS_13TeV=25.25), #2016 TuneCUETP8M1 XSDB value
+            "CrossSection" : XSValues(
+                XS_2016=25.25, XSSource_2016="XSDB",
+                XS_2017=18.18, XSSource_2017="GenXSecAnalyzer",
+                XS_2018=18.21, XSSource_2018="GenXSecAnalyzer",
+            ),
         },
         "QCD_Pt-800to1000_MuEnrichedPt5" : {
-            "CrossSection" : XSValues(XS_13TeV=4.723), #2016 TuneCUETP8M1 XSDB value
+            "CrossSection" : XSValues(
+                XS_2016=4.723, XSSource_2016="XSDB",
+                XS_2017=3.277, XSSource_2017="GenXSecAnalyzer",
+                XS_2018=3.276, XSSource_2018="GenXSecAnalyzer",
+            ),
         },
         "QCD_Pt-1000toInf_MuEnrichedPt5" : {
-            "CrossSection" : XSValues(XS_13TeV=1.613), #2016 TuneCUETP8M1 XSDB value
-        },
-        # QCD pT-hat binned: cross sections from AN2017_013_v17
-        "QCD_Pt-15to7000" : {
-            "CrossSection" : XSValues(XS_13TeV=1372000000.0), #2018 TuneCP5 GenXsecAnalyzer value
+            "CrossSection" : XSValues(
+                XS_2016=1.613, XSSource_2016="XSDB",
+                XS_2017=1.079, XSSource_2017="GenXSecAnalyzer",
+                XS_2018=1.079, XSSource_2018="GenXSecAnalyzer",
+            ),
+        },
+        "QCD_Pt-15to7000_Flat" : {
+            "CrossSection" : XSValues(
+                XS_2016=1976000000.0, XSSource_2016="XSDB",
+                XS_2017=1370000000.0, XSSource_2017="XSDB",
```

Comment:
i'm a bit suspicious that this is the exact value for QCD_FlatPt_15_3000 in 2018, can you double check

Reply:
Depends on if you'd like to use the "Flat" or "Flat2017" sample as a reference:
https://cms-gen-dev.cern.ch/xsdb/?columns=38472128&currentPage=0&ordDirection=1&ordFieldName=accuracy&pageSize=0&searchQuery=process_name%3DQCD%20%26%26%20process_name%3DFlat%20%26%26%20process_name%3D7000%20%26%26%20process_name%3DCP5

Reply:
I created a different section for the Flat2017 sample below, so I'd like to keep this value here.

Reply:
I'm not sure what you mean. It seems suspicious to me to get the same values for 15to3000 and 15to7000.

Reply:
I removed the following entries:
```
        "QCD_FlatPt_15_3000" : {
            "CrossSection" : XSValues(
                XS_2017=1985000000.0, XSSource_2017="XSDB (LO)",
                XS_2018=1370000000.0, XSSource_2018="XSDB (LO)",
            ),
        },
        "QCD_FlatPt_15_3000HS" : {
            "CrossSection" : XSValues(
                XS_13TeV=1356000000.0, XSSource_13TeV="Unknown",
            ),
        },
```
Not only could we not trust their values, but TreeMaker doesn't even contain file lists for these samples. I'd rather these just not pollute the code. These are names for relval samples anyway.

Reply:
Now that we only have 'QCD_Pt-15to7000*' samples I've double checked the values with the GenXSecAnalyzer. They should all make sense now.